components/definitions/newTab.d.ts

```diff
@@ -7,7 +7,8 @@ declare namespace NewTab {
   // Type for custom background and brave background.
   // Custom background uses only image url prop.
   export type BackgroundWallpaper = {
```

Comment:
Optional: You can improve the type definition here to force only one or the other being specified:

```ts
type BackgroundWallpaper = ({ wallPaperSolidColor: string } | { wallpaperImageUrl: string })
    & { commonProperty1?: string, ... };
```

(it might require some updates elsewhere, where it's not sure if a property exists).

Reply:
Yeah, typescript types are pretty fun when you play around with them :D I sometimes fall pretty far down a rabbit hole trying to make mine work just how I want them :laughing: